src/unitxt/operators.py

```diff
@@ -1092,6 +1104,49 @@ def _is_disallowed(self, instance: dict) -> bool:
         return False
 
 
+class FilterByOrder(BaseFilter):
```

Comment:
I think our use of filters is becoming complex, and we are creating a new "programming language" .
Still for this PR, I think it's ok for now to add this.
I suggest calling call it FilterByCondition and not FilterByOrder.

 FilterByCondition(required_values = {"a":4}, condition = "gt") will yield only instances where "a">4

Reply:
Now that I think of it maybe have
FilterByValues just be a special case , where condition is "operator.eq"

Reply:
Changed to FilterByCondition. 
FilterByValues has both required and disallowed values, so it will require two consecutive FilterByCondition instances.